src/Core/src/Handlers/View/ViewHandler.iOS.cs

```diff
@@ -42,14 +42,51 @@ internal static void MapContextFlyout(IElementHandler handler, IContextFlyoutEle
 				if (contextFlyoutContainer.ContextFlyout != null)
 				{
 					if (currentInteraction == null)
-						uiView.AddInteraction(new MauiUIContextMenuInteraction(handler));
+					{
+						if (uiView is WebKit.WKWebView webView)
+						{
+							// If the view is a WKWebView, we need to intercept right-clicks
+							// to show the context menu, so we add a mask view that intercepts
+							// right-clicks and passes them to the context menu interaction.
+							var maskView = new InterceptRightClickWebViewMaskView(uiView.Bounds)
```

Comment:
Right now, every time you re-apply the flyout you’ll stack another `InterceptRightClickWebViewMaskView`. I think here we can use unique Tag to find and remove the interaction and the MaskView from the SuperView.